File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let numSet = new Set(nums);
        let res = 0;
        for (let num of nums) {
            // Check if num is start of chain
            if (numSet.has(num - 1)) continue;
            let cur = 0;
            while (numSet.has(num)) {
                cur += 1;
                num += 1;
            }
            res = Math.max(res, cur);
        }
        return res;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        start, end = 0, 0
        res = 0
        while end < len(prices):
            if prices[end] <= prices[start]:
                start = end
            res = max(res, prices[end] - prices[start])
            end += 1
        
        return res